tensorrt_llm/_torch/modules/fused_moe/moe_prefetch_manager.py

```diff
@@ -0,0 +1,183 @@
+from typing import List, Tuple
+
+import torch
+
+
+class MoEPrefetchManager:
```

Comment:
I don't see anything here that is unique to MOE (i.e no JIT loading). Will it be easy to generalise this to support regular dense layers too?